Resolve issues with JPype1 0.7.2, 0.7.3

In #278, iiasa/message_ix#313, and iiasa/message_data#106, we excluded JPype1 version 0.7.2 from CI builds because it causes segfaults. The builds succeed with version 0.7.1.

We need to investigate and ensure that ixmp (and thus the whole MESSAGEix stack) works with the latest JPype1 at the time of the next release. Otherwise, users installing ixmp etc. from PyPI/conda-forge will run into this problem.

Notes here follow for investigation:
- jpype-project/jpype#605 (issue) and jpype-project/jpype#611 (PR fix) also concern exception handling in JPype; these are ~recent and unreleased~ now released in JPype 0.7.3.
- JPype 0.7.3 caused different test issues in both ixmp and message_ix: see below at https://github.com/iiasa/ixmp/issues/279#issuecomment-619439059. The **temporary** mitigation is to exclude *both* JPype 0.7.2 and 0.7.3.

For **message_ix** (not ixmp), Travis jobs that use JPype 0.7.3 such as 1729.1 are failing [with messages like](https://travis-ci.com/github/iiasa/message_ix/jobs/323308726#L1457):
```
>           self.jindex[ts].readSolutionFromGDX(*args)
E           TypeError: Ambiguous overloads found for at.ac.iiasa.ixmp.objects.MsgScenario.readSolutionFromGDX(str,str,str,java.util.LinkedList,java.util.LinkedList,bool) between:
E           	public void at.ac.iiasa.ixmp.objects.MsgScenario.readSolutionFromGDX(java.lang.String,java.lang.String,java.lang.String,java.util.List,java.util.List,boolean) throws at.ac.iiasa.ixmp.exceptions.IxException
E           	public void at.ac.iiasa.ixmp.objects.Scenario.readSolutionFromGDX(java.lang.String,java.lang.String,java.lang.String,java.util.LinkedList,java.util.LinkedList,boolean) throws at.ac.iiasa.ixmp.exceptions.IxException

../../../miniconda/envs/testing/lib/python3.8/site-packages/ixmp/backend/jdbc.py:346: TypeError
```

FYI @zikolach.

I just checked and the signature of `at…Scenario` on `master` seems not to use LinkedList since recently (iiasa/ixmp_source#285): https://github.com/iiasa/ixmp_source/blob/master/src/main/java/at/ac/iiasa/ixmp/objects/Scenario.java#L1374 —so I guess this is included on some .jar file update on some branch? Would that resolve the issue? Do we need to change LinkedList to List in the Python JDBCBackend code?

This seems to be the common cause of a number of different test failures.

----
For **ixmp** *per se*, Travis jobs that use JPype 0.7.3, such as 1398.1, appear to be failing [with messages like](https://travis-ci.com/github/iiasa/ixmp/jobs/323368163#L2566-L2570).

```
────────────────────────────────────────────────────────────────────────────────

test_core.R:66: failure: the canning problem can be solved

scen$var("z")$lvl not equal to 153.675.

'current' is not an environment

────────────────────────────────────────────────────────────────────────────────
```

Only the two R-related tests fail. We discussed this with @adrivinca in Slack, who said the message wasn't a familiar one.

I am not sure if these R failures 

----

For the moment, I will make quick changes to:
- exclude both JPype 0.7.3 **and** 0.7.2 from CI and requirements on `master` and `2.0.x`—this should allow us to see if the R tests are related—and
- do the same for message_ix.

But we need to diagnose and resolve these issues.

--- a/RELEASE_NOTES.rst
+++ b/RELEASE_NOTES.rst
@@ -1,9 +1,16 @@
 Next release
 ============
 
+Migration notes
+---------------
+
+**Updated dependencies.** The minimum version of JPype is 0.7.5.
+
+
 All changes
 -----------
 
+- `#327 <https://github.com/iiasa/ixmp/pull/327>`_: Bump JPype dependency to 0.7.5.
 - `#298 <https://github.com/iiasa/ixmp/pull/298>`_: Improve memory management in :class:`.JDBCBackend`.
 - `#316 <https://github.com/iiasa/ixmp/pull/316>`_: Raise user-friendly exceptions from :meth:`.Reporter.get` in Jupyter notebooks and other read–evaluate–print loops (REPLs).
 - `#315 <https://github.com/iiasa/ixmp/pull/315>`_: Ensure :meth:`.Model.initialize` is always called for new *and* cloned objects.

--- a/ci/conda-requirements.txt
+++ b/ci/conda-requirements.txt
@@ -2,10 +2,7 @@ click
 codecov
 dask
 graphviz
-
-# Temporary exclusions; see iiasa/ixmp#279
-JPype1 != 0.7.2, != 0.7.3, != 0.7.4
-
+JPype1
 jupyter
 numpydoc
 memory_profiler

--- a/ixmp/tests/test_r.py
+++ b/ixmp/tests/test_r.py
@@ -7,6 +7,11 @@ import sys
 import pytest
 
 
+pytestmark = pytest.mark.skip(
+    reason="Temporarily disabled; see https://github.com/iiasa/ixmp/issues/328"
+)
+
+
 @pytest.fixture
 def r_args(request, tmp_env, test_data_path, tmp_path_factory):
     """Arguments for subprocess calls to R."""

--- a/setup.py
+++ b/setup.py
@@ -9,8 +9,7 @@ with open('README.md', 'r') as f:
     LONG_DESCRIPTION = f.read()
 
 INSTALL_REQUIRES = [
-    # Temporary exclusions; see iiasa/ixmp#279
-    'JPype1 >= 0.7, != 0.7.2, != 0.7.3, != 0.7.4',
+    'JPype1 >= 0.7.5',
     'click',
     'dask[array]',
     'graphviz',

